libs/framework/gtest/src/ScheduledEventTestSuite.cc

```diff
@@ -0,0 +1,695 @@
+
+    // And the event is fired
+    EXPECT_EQ(1, count.load());
+
+    // When waiting to short for the event
+    success = event.waitFor(std::chrono::microseconds{1});
+
+    // Then the return status is false (timeout)
+    EXPECT_FALSE(success);
+
+    // When waiting for the event with a timeout longer than the interval
+    success = event.waitFor(std::chrono::milliseconds{10 + ALLOWED_ERROR_MARGIN_IN_MS});
+
+    // Then the return status is success
+    EXPECT_TRUE(success);
+
+    // And the event is fired again
+    EXPECT_EQ(2, count.load());
+}
+
+TEST_F(ScheduledEventTestSuite, ScheduledEventTimeoutLogTest) {
+    //Given a framework with a log callback that counts the number of warning log messages
+    std::atomic<int> logCount{0};
+    auto logCallback = [](void* handle, celix_log_level_e level, const char*, const char*, int, const char* format, va_list args){
+        std::atomic<int>& count = *static_cast<std::atomic<int>*>(handle);
+        if (level == CELIX_LOG_LEVEL_WARNING) {
+            count.fetch_add(1);
+        }
+        FILE* output = stdout;
+        if (level == CELIX_LOG_LEVEL_FATAL || level == CELIX_LOG_LEVEL_ERROR || level == CELIX_LOG_LEVEL_WARNING) {
+            output = stderr;
+        }
+        fprintf(output, "%s: ", celix_logLevel_toString(level));
+        vfprintf(output, format, args);
+        fprintf(output, "\n");
+    };
+    celix_framework_setLogCallback(fw->getCFramework(), &logCount, logCallback);
+
+
+    //And a scheduled event with an initial delay of 1ms and an interval of 1ms that waits for 1ms in the callback
+    //and remove callback.
+    celix_scheduled_event_options_t opts{};
+    opts.name = "Sleep while Processing and Removing Scheduled Event";
+    opts.initialDelayInSeconds = 0.01;
+    opts.intervalInSeconds = 0.01;
+    opts.callback = [](void*){
+        std::this_thread::sleep_for(std::chrono::milliseconds{200});
+    };
+    opts.removeCallback = [](void*){
+        std::this_thread::sleep_for(std::chrono::milliseconds{200});
+    };
+    long eventId = celix_bundleContext_scheduleEvent(fw->getFrameworkBundleContext()->getCBundleContext(), &opts);
+
+    //When the event is woken up
+    celix_bundleContext_wakeupScheduledEvent(fw->getFrameworkBundleContext()->getCBundleContext(), eventId);
+
+    // Then the log callback is called with a warning log message within an error margin, because callback took too long
+    waitFor([&] { return logCount.load() == 1; },
+            std::chrono::milliseconds{ALLOWED_ERROR_MARGIN_IN_MS + 200});
+    EXPECT_EQ(1, logCount.load());
+
+    //When removing the event
+    celix_bundleContext_removeScheduledEvent(fw->getFrameworkBundleContext()->getCBundleContext(), eventId);
+
+    //Then the log callback is called at least one more time with a warning log message, because callback took too long
+    //(note the logCount can be increased more than once, due to another processing thread)
+    EXPECT_GE(logCount.load(), 2);
```

Comment:
The current implementation does not log if  remove callbacks takes too long.
Either this should be removed, or we should log if the remove callback takes too much time to run.

Reply:
This test whether the an additional log messages is logged during the remove scheduled event (due to the remove callback taking too long).